ensemble-view/src/main/java/hec/ensembleview/controllers/DatabaseController.java

```diff
@@ -0,0 +1,145 @@
+package hec.ensembleview.controllers;
+
+import hec.RecordIdentifier;
+import hec.SqliteDatabase;
+import hec.dss.ensemble.DssDatabase;
+import hec.ensembleview.DatabaseHandlerService;
+import hec.ensembleview.viewpanels.OptionsPanel;
+
+import javax.swing.*;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class DatabaseController {
+    Logger logger = Logger.getLogger(DatabaseController.class.getName());
+    private JPanel filePathPanel;
+    private JComboBox<ZonedDateTime> dateTimes;
+    private JComboBox<RecordIdentifier> locations;
+    private JTextField filePath;
+    private DatabaseHandlerService databaseHandlerService;
+    boolean isNewLoad;  //checks if new database is added. If database is new, set to true
+
+    public DatabaseController(OptionsPanel optionsPanel) {
+        initiateDatabaseListener(optionsPanel);
+    }
+
+    private void initiateDatabaseListener(OptionsPanel optionsPanel) {
+        optionsPanel.setDatabaseListener(new DatabaseListener() {
+            @Override
+            public void initialize(JPanel filePathPanel, JComboBox<ZonedDateTime> dateTimes, JComboBox<RecordIdentifier> locations, JTextField filePath) {
```

Comment:
Can the access modifier for most of these setter methods be changed from public to private?

Reply:
Changed to private